orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/DetermineRollbackCandidatesTask.java

```diff
@@ -105,172 +105,120 @@ public long getTimeout() {
-                          "onlyEnabledServerGroups", Boolean.FALSE))
-                  .booleanValue();
-
-      ImageDetails imageDetails =
-          previousImageRollbackSupport.getImageDetailsFromEntityTags(
-              stageData.cloudProvider,
-              stageData.credentials,
-              region,
-              newestEnabledServerGroupInRegion.name);
-
-      RollbackDetails rollbackDetails = null;
-      if (imageDetails != null) {
-        // check for rollback candidates based on entity tags
-        logger.info(
-            "Looking for rollback candidates in cluster {}, region {} based on entity tags, "
-                + (onlyEnabled ? "excluding" : "including")
-                + " disabled server groups",
-            moniker.get().getCluster(),
-            region);
-
-        rollbackDetails =
-            fetchRollbackDetails(
-                imageDetails,
-                newestEnabledServerGroupInRegion,
-                onlyEnabled ? enabledServerGroupsInRegion : allServerGroupsInRegion);
-      }
+      ServerGroup serverGroupToRollBack = getServerGroupToRollBack(enabledServerGroupsInRegion);
+
+      RollbackDetails candidateDetails =
+          findBestCandidate(
+              allServerGroupsInRegion,
+              enabledServerGroupsInRegion,
+              serverGroupToRollBack,
+              stageData,
+              cluster,
+              region);
+
+      logger.info(
+          "Found rollback candidate in cluster {}, region {}: {}",
+          cluster,
+          region,
+          candidateDetails.rollbackContext.get("restoreServerGroupName"));
+
+      imagesToRestore.add(getImageToRestore(region, candidateDetails));
+      rollbackTypes.put(region, candidateDetails.rollbackType.toString());
+      rollbackContexts.put(
+          region,
+          getRollbackContext(
+              stageData.targetHealthyRollbackPercentage, serverGroupToRollBack, candidateDetails));
+    }
 
-      if (rollbackDetails == null) {
-        // check for rollback candidates based on previous server groups
-        logger.info(
-            "Looking for rollback candidates in cluster {}, region {} based on previous server groups, "
-                + (onlyEnabled ? "excluding" : "including")
-                + " disabled ones",
-            moniker.get().getCluster(),
-            region);
-
-        rollbackDetails =
-            fetchRollbackDetails(
-                newestEnabledServerGroupInRegion,
-                onlyEnabled ? enabledServerGroupsInRegion : allServerGroupsInRegion);
-      }
+    return buildResult(imagesToRestore, rollbackTypes, rollbackContexts);
+  }
 
-      if (rollbackDetails != null) {
-        logger.info(
-            "Found rollback candidate in cluster {}, region {}: {}",
-            moniker.get().getCluster(),
-            region,
-            rollbackDetails.rollbackContext.get("restoreServerGroupName"));
+  private RollbackDetails findBestCandidate(
```

Comment:
Even though this function doesn't contain much code, I pulled it out because my next PR should hopefully just touch this function without changing anything else.